app/views/components/_option-select.html.erb

```diff
@@ -3,20 +3,31 @@
   checkboxes_id = "checkboxes-#{SecureRandom.hex(4)}"
   checkboxes_count_id = checkboxes_id + "-count"
   show_filter ||= false
+%>
 
-  if show_filter
-    filter_element = ActionController::Base.new.render_to_string(:partial => "govuk_publishing_components/components/input", :locals => {
-      label: {
-        text: "Filter " + title
-      },
-      name: "option-select-filter",
-      controls: checkboxes_id,
-      describedby: checkboxes_count_id
-    })
+<% if show_filter %>
+  <%
+    filter_id ||= "input-#{SecureRandom.hex(4)}"
```

Comment:
Can this be defined at the top of the file?

Reply:
I put it in here to save a tiny bit of processing - unless there is a filter, we don't need the `filter_id` variable.

Reply:
Good point, sorry I think I misread and didn't realise this was the option select component, not a filter component 🤦‍♀️